src/main/resources/assets/paths/darV3ByReferenceId.yaml

```diff
@@ -0,0 +1,31 @@
+put:
+  summary: Update a Data Access Request by Reference Id
+  description: |
+    Updates a updated Data Access Request by Reference Id
+    Datasets are restricted to those for which the user has a Library Card with a Data Access Agreement for the dataset's Data Access Committee
+  parameters:
+    - name: referenceId
+      in: path
+      description: The referenceId of the Data Access Request
+      required: true
+      schema:
+        type: string
+    - name: dar
+      in: body
+      description: The fields that represent a DAR in json format.
+      required: true
+      schema:
+        $ref: '../schemas/DataAccessRequest.yaml'
+  tags:
+    - Data Access Request
+  responses:
```

Comment:
Again, do we need to add a 400 response for the error thrown if the user does not have all of the DAAs on their LC for any of the datasets in this update to the DAR?

Also, I'm realizing that in the update case, we are going to make the dataset selection in the DAR form readonly after initial creation, meaning the dataset selection will never change in the update case. That work is yet to be implemented, but is covered in [this ticket](https://broadworkbench.atlassian.net/browse/DCJ-48). So, will we ever run into the case where we have to enforce the DAA restrictions in this endpoint?

Reply:
Also, I don't actually think we created this new endpoint in `DataAccessRequestResource`.

Reply:
This is a great catch and points out something we sometimes hit. I'm going to remove this and file a ticket for removing the v2 version of this endpoint. This is a PUT on a submitted DAR which is not something we should be allowing clients to do. The v2 API is not in use in the UI.

Reply:
https://broadworkbench.atlassian.net/browse/DCJ-400